crates/services/tx_status_manager/src/service.rs

```diff
@@ -0,0 +1,137 @@
+use fuel_core_services::{
+    RunnableService,
+    RunnableTask,
+    ServiceRunner,
+    StateWatcher,
+    TaskNextAction,
+};
+use fuel_core_types::services::{
+    p2p::{
+        GossipData,
+        PreConfirmationMessage,
+        Sealed,
+    },
+    preconfirmation::Preconfirmation,
+    txpool::TransactionStatus,
+};
+use futures::StreamExt;
+
+use crate::{
+    config::Config,
+    manager::TxStatusManager,
+    ports::{
+        P2PPreConfirmationGossipData,
+        P2PPreConfirmationMessage,
+        P2PSubscriptions,
+    },
+    subscriptions::Subscriptions,
+    update_sender::TxStatusChange,
+};
+
+pub struct Task {
+    manager: TxStatusManager,
+    subscriptions: Subscriptions,
+}
+
+impl Task {
+    fn new_preconfirmations_from_p2p(
+        &mut self,
+        preconfirmations: P2PPreConfirmationMessage,
+    ) {
+        match preconfirmations {
+            PreConfirmationMessage::Delegate(_sealed) => {
+                // TODO[RC]: Handle the delegate message,
+            }
+            PreConfirmationMessage::Preconfirmations(sealed) => {
+                let Sealed {
+                    signature: _,
+                    entity,
+                } = sealed;
+                // TODO[RC]: Add signature verification
+                entity.preconfirmations.into_iter().for_each(
+                    |Preconfirmation { tx_id, status }| {
+                        let status: TransactionStatus = status.into();
+                        self.manager.status_update(tx_id, status);
+                    },
+                );
+            }
+        }
```

Comment:
We need to create a follow-up issue to implement verification logic and put a todo here with the link, we don't need to implement it right now=)